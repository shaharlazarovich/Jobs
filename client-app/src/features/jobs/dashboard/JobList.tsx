import React, { useContext, Fragment } from 'react'
import { Item, Label } from 'semantic-ui-react'
import { observer } from 'mobx-react-lite'
import JobListItem from './JobListItem'
import { RootStoreContext } from '../../../app/stores/rootStore'
import {format} from 'date-fns'

const JobList: React.FC = () => {
    const rootStore = useContext(RootStoreContext);
    const {jobsByDate} = rootStore.jobStore; //destructure of parameters
    return ( //we want to group our jobs by date, using the method we created in jobStore
    <Fragment>
        {jobsByDate.map(([group, jobs])=> (
            <Fragment key={group}>
                <Label size='large' color='blue'>
                    {format(new Date(group), 'eeee do MMMM')}
                </Label>
                <Item.Group divided>
                {jobs.map(job => (
                    <JobListItem key={job.id} job={job} />
                ))}
                </Item.Group>
            </Fragment>
        ))}
    </Fragment>
    )
}

export default observer(JobList)
