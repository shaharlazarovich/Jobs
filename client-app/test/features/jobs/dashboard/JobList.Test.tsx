import { shallow } from "enzyme";
import React from "react";
import { findByTestAttr } from '../../../common/testUtils';
import JobList from "../../../../src/features/jobs/dashboard/JobList";

const setup = (props={}) => {
    const wrapper = shallow(<JobList {...props} />)
    return wrapper;
}

describe('Job List UniTest', () => {
    let wrapper:any;
    beforeEach(() => {
        wrapper = setup([]);
    });
    test('renders without error', () => {
        const component = findByTestAttr(wrapper, 'component-job-list');
        expect(component.length).toBe(1);
    });
})