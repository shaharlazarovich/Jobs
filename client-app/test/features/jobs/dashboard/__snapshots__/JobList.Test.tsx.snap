// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job List UniTest should match the snapshot 1`] = `null`;
